packages/paste-core/components/combobox/src/index.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {useUID} from 'react-uid';
+import {useComboboxPrimitive} from '@twilio-paste/combobox-primitive';
+import {ChevronDownIcon} from '@twilio-paste/icons/esm/ChevronDownIcon';
+import {Box} from '@twilio-paste/box';
+import {FormHelpText, FormInput, FormLabel} from '@twilio-paste/form';
+import {Text} from '@twilio-paste/text';
+
+export interface ComboboxProps {
+  autocomplete?: boolean;
+  disabled?: boolean;
+  hasError?: boolean;
+  helpText?: string | React.ReactNode;
+  id?: string;
+  items: string[];
+  labelText: string | NonNullable<React.ReactNode>;
+  name?: string;
+  placeholder?: string;
+  required?: boolean;
+}
+
+const ComboboxIconWrapper: React.FC<{children: React.ReactNode}> = ({children}) => (
+  <Box
+    alignItems="center"
+    display="inline-flex"
+    position="absolute"
+    pointerEvents="none"
+    right="space40"
+    top="50%"
+    transform="translateY(-50%)"
+    zIndex="zIndex10"
+  >
+    {children}
+  </Box>
+);
+
+/* eslint-disable react/no-array-index-key */
+const Combobox: React.FC<ComboboxProps> = ({
+  autocomplete,
+  disabled,
+  helpText,
+  hasError,
+  id,
+  items,
+  labelText,
+  name,
+  placeholder,
+  required,
+}) => {
+  const [inputItems, setInputItems] = React.useState(items);
+  const {
+    getComboboxProps,
+    getInputProps,
+    getItemProps,
+    getLabelProps,
+    getMenuProps,
+    getToggleButtonProps,
+    highlightedIndex,
+    isOpen,
+  } = useComboboxPrimitive({
+    items: inputItems,
+    onInputValueChange: ({inputValue}) => {
+      if (autocomplete && inputValue) {
+        setInputItems(items.filter(item => item.toLowerCase().startsWith(inputValue.toLowerCase())));
+      }
+      return inputValue;
+    },
+  });
+  const helpTextId = useUID();
+  return (
+    <>
+      <FormLabel disabled={disabled} htmlFor={id} required={required} {...getLabelProps()}>
+        {labelText}
+      </FormLabel>
+      <Box {...getComboboxProps({role: 'combobox'})} display="flex" position="relative">
+        <FormInput
+          {...getInputProps()}
+          {...getToggleButtonProps({tabIndex: 0})}
+          disabled={disabled}
+          hasError={hasError}
+          id={id}
+          name={name}
+          placeholder={placeholder}
+          required={required}
+          type="text"
```

Comment:
What if I wanna do insertAfter and insertBefore? Like selects?

Reply:
`insertBefore` works, but `insertAfter` complicates things since that is part of the `FormInput` we have overlaps with the chevron:

![image](https://user-images.githubusercontent.com/1350081/82096422-2f50d280-96be-11ea-8084-cbfd5d5fd36e.png)

Unless we create another prefix and suffix for combobox, or export those from the form package. What do you think?

Reply:
Checkout what @andipants12 did in select https://github.com/twilio-labs/paste/blob/master/packages/paste-core/components/form/src/select/Select.tsx